src/main/java/mate/academy/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,173 @@
+            deleteProductsFromDB(connection, shoppingCart.getId());
+            addProductsToDB(connection, shoppingCart);
+            PreparedStatement statement = connection.prepareStatement("UPDATE shopping_carts "
+                    + "SET user_id = ? "
+                    + "WHERE cart_id = ? AND deleted = FALSE");
+            statement.setLong(1, shoppingCart.getUserId());
+            statement.setLong(2, shoppingCart.getId());
+            statement.executeUpdate();
+            return shoppingCart;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update shopping cart", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement("UPDATE shopping_carts SET deleted = TRUE WHERE cart_id = ?");
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't delete cart with id" + id, ex);
+        }
+    }
+
+    private void addProductsToDB(Connection connection,
+                                 ShoppingCart shoppingCart) throws SQLException {
+        PreparedStatement statement = connection
+                .prepareStatement("INSERT INTO shopping_carts_products "
+                        + "(cart_id, product_id) VALUES (?, ?)");
+        statement.setLong(1, shoppingCart.getId());
+        for (Product product : shoppingCart.getProducts()) {
+            statement.setLong(2, product.getId());
+            statement.executeUpdate();
+        }
+    }
+
+    private void deleteProductsFromDB(Connection connection,
+                                      Long id) throws SQLException {
+        PreparedStatement statement = connection
+                .prepareStatement("DELETE FROM shopping_carts_products WHERE cart_id = ?");
+        statement.setLong(1, id);
+        statement.executeUpdate();
+    }
+
+    private static ShoppingCart getCartFromResultSet(ResultSet resultSet) throws SQLException {
+        long cartId = resultSet.getLong("cart_id");
+        long userId = resultSet.getLong("user_id");
+        ShoppingCart shoppingCart = new ShoppingCart(userId);
+        shoppingCart.setId(cartId);
+        shoppingCart.setProducts(getProductsFromShoppingCart(cartId));
+        return shoppingCart;
+    }
+
+    private static List<Product> getProductsFromShoppingCart(long cartId) {
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement("SELECT * FROM products p "
+                    + "INNER JOIN shopping_carts_products scp ON p.product_id = scp.product_id "
+                    + "WHERE cart_id = ?;");
```

Comment:
Sometimes you use ; at the end of SQL query, and sometimes you don't. It would look nicer if the format of SQL requests was consistent)

Reply:
ok